### PR TITLE
Re-instate documentation for lineprofiler and notebook plugins

### LINKED ISSUE #324: Docs for spyder-lineprofiler are hidden
# Issue Report

## Issue Description

The documentation for the `spyder-lineprofiler` plugin are hidden, even though the plugin is now compatible with Spyder 5.

### Where is the relevant portion of the documentation located?

N/A

### What would you suggest it be replaced with?

We can just use the old docs.

### Other information

* Is the problem specific to documentation for a particular version of Spyder?

    * [ ] N/A
    * [x] Spyder 5 (``master``)
    * [ ] Spyder 4 (``4.x``)
    * [ ] Spyder 3 (``3.x``, frozen)

* Are you planning on submitting a Github pull request with a suitable change?

    * [x] Yes
    * [ ] No
    * [ ] Maybe so

### PR DESCRIPTION
**Pull request by jitseniesen**
# Pull Request

## Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder-docs/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 4.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed

## Description of Changes

Remove the warning in the documentation for the `spyder-lineprofiler` and `spyder-notebook` plugins that they are incompatible with Spyder 5, because this is no longer the case. Also include them both under "Plugins" in the left sidebar.

This reverts PRs #283 and #305.

### Issue(s) Resolved

Fixes #324

### CODE REVIEW

File: doc/plugins/index.rst [hunk 1]
```diff
@@ -5,4 +5,5 @@ Spyder Plugins
 .. toctree::
     :maxdepth: 2
 
+    lineprofiler
     terminal
```
Comment (CAM-Gerlach) [inline doc/plugins/index.rst:9]:
```suggestion
    :maxdepth: 2
    :glob:

    *
```

**Review by CAM-Gerlach — COMMENTED** [review]
Thanks @jitseniesen ; looks good, but since @ccordoba12 mentioned he would be releasing the final update today to make Spyder Notebook fully compatible with Spyder 5, could you just make the toctree a glob like the others to automatically , and removing the outdated warning and `:orphan:` tag for the Spyder Notebooks doc as well? I made a suggestion to do the former.

Also, are the text and screenshots still up to date for the current version of the plugin? If not, should we add an admonition mentioning this at the top of the docs, so as to not confuse users? Thanks.

**Comment by jitseniesen** [comment]
> [...] he would be releasing the final update today to make Spyder Notebook fully compatible with Spyder 5, could you just make the toctree a glob like the others to automatically, and removing the outdated warning and `:orphan:` tag for the Spyder Notebooks doc as well?

No problem, I will wait until spyder-notebook is released and re-instate its docs.

> Also, are the text and screenshots still up to date for the current version of the plugin?

They are still up to date for the line profiler plugin, but probably not for the notebook plugin. I will check when it is released.

**Comment by CAM-Gerlach** [comment]
Hey @ccordoba12 , any update on the timeline for Spyder-Notebooks?

@jitseniesen , if its going to delay this significantly or we want to update the screenshots before going live with the new docs, we could just go ahead and merge this and then we can implement the change for Spyder-Notebooks separately, if you think that's better.

**Comment by ccordoba12** [comment]
> Hey @ccordoba12 , any update on the timeline for Spyder-Notebooks?

Sorry for the delay, I just released Spyder-notebook 0.4.0. I expect for it to be available in conda-forge later today or tomorrow.

**Comment by CAM-Gerlach** [comment]
@jitseniesen seems like @ccordoba12 just released Spyder-Notebook 0.4.0, FYI, and at the meeting he stated that he didn't believe there were substantial UI changes that would impact the screenshots, at least of the plugins themselves (though Spyder in the background will look different, of course).

**Comment by jitseniesen** [comment]
Docs for the notebook plugin are now also revived.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM. thanks @jitseniesen !

**Pull request merged**